### PR TITLE
Added table of contents and refactored registers

### PR DESCRIPTION
**Pull request by Dishebh**
#### Changes done:
- [x] Added table of contents and refactored `digital registers`.

Preview Link: https://deploy-preview-317--tender-ardinghelli-ebb79f.netlify.app/docs/registers/ss.html
Preview Link: https://deploy-preview-317--tender-ardinghelli-ebb79f.netlify.app/docs/registers/sp.html
Preview Link: https://deploy-preview-317--tender-ardinghelli-ebb79f.netlify.app/docs/registers/ps.html
Preview Link: https://deploy-preview-317--tender-ardinghelli-ebb79f.netlify.app/docs/registers/pp.html

#### ✅️ By submitting this PR, I have verified the following

- [x] Checked to see if a similar PR has already been opened 🤔️
- [x] Reviewed the contributing guidelines 🔍️
- [x] Sample preview link added (add a link from the checks tab after checks complete)
- [x] Tried Squashing the commits into one

### CODE REVIEW

**Comment by Ananthavijay** [comment]
Compare this one and https://github.com/CircuitVerse/Interactive-Book/pull/210 and then merge one

**Comment by Dishebh** [comment]
> Compare this one and #210 and then merge one

I guess both target different changes and commits. I committed `toc` and some docs changes (not specifically targeted to remove plagiarism).

File: docs/registers/pp.md [hunk 1]
```diff
@@ -9,15 +9,19 @@ comments: true
...
+{: .no_toc .text-delta }
 
+1. TOC
+{:toc}
 
 
 ## Introduction
  
-In this mode, the 4 bit binary input B0, B1, B2, B3 is applied to the data inputs D0, D1, D2, D3 respectively of the four flip-flops. 
+In this mode, the 4-bit binary input B0, B1, B2, B3 is applied to the data inputs D0, D1, D2, D3 respectively of the four flip-flops.
```
Comment (YashKumarVerma) [inline docs/registers/pp.md:21]: content still from this source after refactor
https://www.tutorialspoint.com/computer_logical_organization/pdf/digital_registers.pdf

File: docs/registers/ps.md [hunk 2]
```diff
@@ -9,31 +9,35 @@ comments: true
...
 
-* The binary input word B0, B1, B2, B3 is applied though the same combinational circuit.
+* The binary input word B0, B1, B2, B3 is applied through the same combinational circuit.
 
 * There are two modes in which this circuit can work namely - shift mode or load mode.
 
 ## Load mode
 When the shift/load bar line is low (0), the AND gate 2, 4 and 6 become active they will pass B1, B2, B3 bits to the corresponding flip-flops. 
-On the low going edge of clock, the binary input B0, B1, B2, B3 will get loaded into the corresponding flip-flops. 
+On the low going edge of the clock, the binary input B0, B1, B2, B3 will get loaded into the corresponding flip-flops.
```
Comment (YashKumarVerma) [inline docs/registers/ps.md:33]: content from http://www.brainheaters.in/uploads/2/6/0/3/26031536/adc.pdf

File: docs/registers/pp.md (hunk 1 above)
Comment (Dishebh) [reply docs/registers/pp.md:21]: Can I refer the sources which are listed in the references, and take content from there?

File: docs/registers/ps.md (hunk 2 above)
Comment (Dishebh) [reply docs/registers/ps.md]: I have made some changes to the docs. Please review them as well

**Comment by Dishebh** [comment]
Made some changes. See if they are fine @YashKumarVerma ?

**Pull request merged**